samples/52.teams-messaging-extensions-search-auth-config/src/main/java/com/microsoft/bot/sample/teamssearchauth/TeamsMessagingExtensionsSearchAuthConfigBot.java

```diff
@@ -188,23 +189,25 @@ public TeamsMessagingExtensionsSearchAuthConfigBot(
             .thenCompose(response -> {
                 if (response == null || StringUtils.isEmpty(response.getToken())) {
                     // There is no token, so the user has not signed in yet.
+
+                    MessagingExtensionResponse extensionResponse = new MessagingExtensionResponse();
+                    CardAction cardAction = new CardAction();
+                    cardAction.setType(ActionTypes.OPEN_URL);
+                    cardAction.setValue(extensionResponse);
+                    cardAction.setTitle("Bot Service OAuth");
+
+                    MessagingExtensionSuggestedAction suggestedAction = new MessagingExtensionSuggestedAction();
+                    suggestedAction.setAction(cardAction);
+
+                    MessagingExtensionResult result = new MessagingExtensionResult();
+                    result.setType("auth");
+                    result.setSuggestedActions(suggestedAction);
+
+
+                    extensionResponse.setComposeExtension(result);
+
                     return tokenProvider.getOAuthSignInLink(turnContext, connectionName)
-                        .thenApply(link -> new MessagingExtensionResponse() {{
-                            setComposeExtension(new MessagingExtensionResult() {{
-                                setType("auth");
-                                setSuggestedActions(
-                                    new MessagingExtensionSuggestedAction() {{
-                                        setAction(
-                                            new CardAction() {{
-                                                setType(ActionTypes.OPEN_URL);
-                                                setValue(link);
-                                                setTitle("Bot Service OAuth");
-                                            }}
-                                        );
-                                    }}
-                                );
-                            }});
-                        }});
+                        .thenApply(link -> extensionResponse);
```

Comment:
I think that this will be better if we do the initialization in the `thenApply` instead of outside of it.

Something like:
```java
.thenApply(link -> {
    MessagingExtensionResponse extensionResponse = new MessagingExtensionResponse();
    CardAction cardAction = new CardAction();
    cardAction.setType(ActionTypes.OPEN_URL);
    cardAction.setValue(extensionResponse);
    cardAction.setTitle("Bot Service OAuth");

    MessagingExtensionSuggestedAction suggestedAction = new MessagingExtensionSuggestedAction();
    suggestedAction.setAction(cardAction);

    MessagingExtensionResult result = new MessagingExtensionResult();
    result.setType("auth");
    result.setSuggestedActions(suggestedAction);
    extensionResponse.setComposeExtension(result);

    return extensionResponse;
});
```